### ISSUE TITLE
[Error] Ubuntu 23.04 guest has no network and no sudo

### ISSUE THREAD

**Issue by pettermk** [89luca89/distrobox#810]
Hi,

I'm trying to make Ubuntu 23.04 to work as a guest, on hosts Ubuntu 22.04 and Fedora 38, and facing multiple issues.

First, the stock 23.04 images come preconfigured with a user called ubuntu with uid 1000. This conflicts with how users are setup and the container ends up in a state where it is not possible to `sudo`, although the container does work to some extent. I have made pull request for this one if you'd like to apply my fix https://github.com/89luca89/distrobox/pull/807

Second, I am facing DNS issues. In the container, `/etc/resolv.conf` is empty, and the first time I saw it I could successfully
```
sudo rm /etc/resolv.conf
sudo ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
```
and the container was happily connected. However this setup has been flaky and one of my guests is now in a state where this file is empty again and cannot be removed. Unsure what would be a good way to fix this permanently.

Outside of the 23.04 snags (which largely seem to stem from ubuntu changes), I'm loving distrobox and thanks for all your hard work!

**Comment by k1mbl3** [comment]
I had apparently the same problem with dns using the stable version (installed with curl -s https://raw.githubusercontent.com/89luca89/distrobox/main/install | sudo sh).

Using Pop Os 22.04 as host and Ubuntu 22.10 as guest, the guest was failing to resolve any dns query.

Downgrading to 1.4.2.1 (available on ppa:michel-slm/distrobox) made my ubuntu guest come back to life

**Comment by alcir** [comment]
Which version are you using?
`distrobox version`
 I had a similar issue with resolv.conf, but it was eventually solved with the latest release.

**Comment by pettermk** [comment]
I did another round of testing, and I see that distrobox < 1.5.0 has a problem installing the packages on ubuntu, as there was one that was no longer available on 23.04, and the latest versions are failing on DNS and user setup (no `sudo` as reported). I see the same as @rrahl is, and it is related to how /etc/resolv.conf gets created.

Looks like it correctly detects the symlink, but I believe it should point to `/run/host/etc/../run/systemd/resolve/stub-resolv.conf` instead of `/run/host/../run/systemd/resolve/stub-resolv.conf`, so I'm wondering if things start to go wrong here https://github.com/89luca89/distrobox/blob/main/distrobox-init#L1383

**Comment by 89luca89** [comment]
This has been now fixed by #807
